packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -130,12 +137,25 @@ function computeRequestTracingInfo(request: RequestCompleteEvent) {
     _dd: {
       span_id: request.spanId!.toDecimalString(),
       trace_id: request.traceId!.toDecimalString(),
+      rule_psr: configuration.tracingSampleRate / 100, // rule_psr needs to be between 0-1
     },
   }
 }
 
-function computeEntryTracingInfo(entry: RumPerformanceResourceTiming) {
-  return entry.traceId ? { _dd: { trace_id: entry.traceId } } : undefined
+function computeEntryTracingInfo(entry: RumPerformanceResourceTiming, configuration: RumConfiguration) {
+  if (entry.traceId) {
+    return {
+      _dd: {
+        trace_id: entry.traceId,
+        rule_psr: configuration.tracingSampleRate / 100, // rule_psr needs to be between 0-1
+      },
+    }
+  }
+  return {
+    _dd: {
+      rule_psr: configuration.tracingSampleRate / 100, // rule_psr needs to be between 0-1
+    },
+  }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think this would be fine:
```suggestion
  return {
    _dd: {
      trace_id: entry.traceId,
      rule_psr: configuration.tracingSampleRate / 100, // rule_psr needs to be between 0-1
    },
  }
```

If `_dd.trace_id` is undefined, it will be skipped during JSON serialization.

Reply:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏in `computeRequestTracingInfo`, we define `rule_psr` only if the request have been traced. Shouldn't we do the same here? Or should it be changed in `computeRequestTracingInfo`?

Reply:
Good point! I think yes, furthermore I think the condition of this check should be lifted to `processRequest` and `processResourceEntry`